src/contract.rs

```diff
@@ -1294,14 +1346,59 @@ fn process_entries_and_exits(deps: DepsMut, env: Env) -> Result<Vec<CosmosMsg>,
     )?;
 
     let mut messages: Vec<CosmosMsg> = vec![];
+
     let mut total_dollars_in_vault = Decimal::zero();
     let mut address_dollars_map: HashMap<Addr, Decimal> = HashMap::new();
 
+    let mut uncompounded_rewards: Vec<Coin> = UNCOMPOUNDED_REWARDS.load(deps.storage)?;
+
+    // add the uncompounded commission to the rest of the commission and deduct them from the rewards
+    if let Some(commission) = config.commission {
```

Comment:
If for some reason you update the config to remove the commission, your current rewards would get lost. I'd change this to:

let mut_commission_rewards = COMMISSION_REWARDS.load(deps.storage)?;
if COMMISSION_REWARDS != vec![
                coin(0, config.asset0.denom.clone()),
                coin(0, config.asset1.denom.clone()),
            ] {
           ...
           }